docs/training-debug/index.txt

```diff
@@ -1,343 +1,395 @@
 
-.. code:: bash
+.. _step9:
 
-   det experiment create myconfig.yaml my_model_dir -f
+***************************************************
+ Step 9 - Verify that a multi-GPU experiment works
+***************************************************
 
-**How to diagnose failures:** If Step 7 worked but Step 8 does not, there are a few high-level
-categories of issues to check for:
+This step is similar to :ref:`Step 8 <step8>`, except that it introduces distributed training. This
+step only applies if you have multiple GPUs and want to use distributed training.
 
--  Does the error happen when the experiment config has ``searcher.source_trial_id`` set? One thing
-   that can occur in a real experiment that does not occur in a ``--test`` experiment is the loading
-   of a previous checkpoint. Errors when loading from a checkpoint can be caused by architecture
-   change, where the new model code is not architecture-compatible with the old model code.
+#. Configure your system the same as :ref:`Step 7 <step7>`:
 
--  Generally, issues at this step are caused by doing training and evaluation continuously, so focus
-   on how that change may cause issues with your code.
+   Set ``resources.slots_per_trial`` to a number greater than ``1``. For example:
 
-9. Does a multi-GPU experiment work?
-====================================
+   .. code:: yaml
 
-This step is like Step 8 except that it introduces distributed training. Naturally, this step is
-only relevant if you have multiple GPUs and you wish to use distributed training.
+      resources:
+        slots_per_trial: 2
 
-**How to test:** Configuration should be like Step 7, except you will now set
-``resources.slots_per_trial`` to some number greater than 1:
+#. Create your experiment:
 
-.. code:: yaml
+   .. code:: bash
 
-   resources:
-     slots_per_trial: 2
+      det experiment create myconfig.yaml my_model_dir -f
 
-Then create your experiment:
+#. Diagnose failures:
 
-.. code:: bash
+   If you are using the ``determined`` library APIs correctly, distributed training should work
+   without error. Otherwise, common problems might be:
 
-   det experiment create myconfig.yaml my_model_dir -f
+   -  If your experiment is not being scheduled on the cluster, ensure that the ``slots_per_trial``
+      setting is valid for your cluster. For example:
 
-**How to diagnose failures:** If you are using the ``determined`` library APIs correctly, then
-theoretically distributed training should "just work". However, you should be aware of some common
-pitfalls:
+      -  If you have four Determined agents running with four GPUs each, your ``slots_per_trial``
+         could be ``1``, ``2``, ``3``, or ``4``, which would fit on a single machine.
+      -  A ``slots_per_trial`` value of ``8``, ``12``, or ``16`` would completely utilize a number
+         of agent machines.
+      -  A ``slots_per_trial`` value of ``5``, implies more than one agent but is not a multiple of
+         agent size.
```

Comment:
this is an invalid case, but that is not explained in the text of the bullet

Reply:
Clarified as requested, along with the last bullet for the '32' case.